apps/src/code-studio/components/ModelManagerDialog.jsx

```diff
@@ -67,21 +101,47 @@ export default class ModelManagerDialog extends React.Component {
   };
 
   handleChange = e => {
-    this.setState({selectedModelId: e.target.value});
+    const model = this.getModelById(e.target.value);
+    this.setState({selectedModel: model});
+  };
+
+  showDeleteConfirmation = () => {
+    this.setState({confirmDialogOpen: true});
+    this.closeModelManager();
+  };
+
+  closeConfirmDialog = () => {
+    this.setState({confirmDialogOpen: false});
+    this.closeModelManager();
+  };
+
+  deleteModel = () => {
+    $.ajax({
+      url: `/api/v1/ml_models/${this.state.selectedModel.id}`,
+      method: 'DELETE'
+    }).then(response => {
+      this.setState({
+        deletionStatus: response.status,
+        confirmDialogOpen: false
```

Comment:
Similar to Brendan's comment about the 'No' button, I wonder if the user should be redirected to the beginning of the dialog prompt after deleting a model. This way, they can visually confirm it's deleted and they can perform another action without having to go back to the settings.